EIPS/eip-223.md

```diff
@@ -0,0 +1,368 @@
+---
+eip: 223
+title: 223 Token with communication model
+description: Token with event handling and communication model
+author: Dexaran (@Dexaran)
+discussions-to: https://ethereum-magicians.org/t/erc-223-token-standard/12894
+status: Review
+type: Standards Track
+category: ERC
+created: 2017-05-03
+---
+
+## Abstract
+
+The following describes an interface for fungible tokens that supports a `tokenReceived` callback to notify contract recipients when tokens are received.
+
+## Motivation
+
+This token introduces a communication model for contracts that can be utilized to straighten the behavior of contracts that interact with such tokens. Specifically, this proposal:
+
+1. Informs receiving contracts of tokens, as opposed to [ERC-20](./eip-20.md) where the recipient of a token transfer gets no notification.
+2. Is more gas-efficient when depositing tokens to contracts.
+3. Allows for `_data` recording for financial transfers.
+
+
+## Specification
```

Comment:
I still would prefer this to be an extension of ERC-20 but it's fine if you don't want that.

Reply:
The main reason why I don't want it is that I'm intentionally overwriting the logic of how `transfer()` functions work.
It is not possible to make it a part of ERC20 standard anyways.

It was described here: https://github.com/ethereum/eips/issues/223

I'm doing it to solve the problem of lost funds i.e. there is no possibility to REJECT incorrectly sent ERC20 tokens due to how its `transfer()` function works and as the result there are millions of dollars lost in ERC20 tokens in ETH and this continues even right now.

Reply:
At the moment of ERC223 creation the whole process was different.

I would like to finalise it as ERC223 which was originally designed as an "alternative standard". Also it would help to avoid confusion for third parties in my opinion.

Reply:
This seems to me to be much more of an extension to ERC-20, and as you said, the process was much different (and extensions were not commonly used). I am okay with this EIP being not an extension with the following two conditions:

1. The definitions for balanceOf, name, synbol, decimals, and all other applicable methods state that their functionality is equivalent to that of ERC-20.
2. The backward compatibility section also states the above.

Reply:
1. Definitions of all functions except `transfer` now refer to ERC-20 analogues.

2. Backwards compatibility section added https://github.com/Dexaran/EIPs/commit/6093c8f32b4a33acd700757776023602e02e0da3